accounts/abi/bind/bind.go

```diff
@@ -224,12 +232,30 @@ func Bind(types []string, abis []string, bytecodes []string, fsigs []map[string]
 		_, ok := isLib[types[i]]
 		contracts[types[i]].Library = ok
 	}
-	// Generate the contract template data content and render it
-	data := &tmplData{
-		Package:   pkg,
-		Contracts: contracts,
-		Libraries: libs,
-		Structs:   structs,
+
+	var data interface{}
+	var templateSource string
+
+	// Generate the contract template data according to contract type (precompile/non)
+	if isPrecompile {
+		if lang == LangJava {
+			return "", errors.New("java binding for precompiled contracts is not supported yet")
+		}
```

Comment:
Should we change this to check to ensure that it is always golang? There's another option for ObjC that would not be caught here and we only support golang anyways.

Reply:
I think that check is a no-op since we statically set it in the precompilegen. but right better to add a not golang check instead java check
https://github.com/ava-labs/subnet-evm/blob/precompile-gen/cmd/precompilegen/main.go#L91